### PR TITLE
Support reading/writing nested containers for ndjson

### PR DESCRIPTION
**Pull request by mattnibs**
Use the shiny new zval.Builder to efficienlty zvals from incoming
json.

Also:
- Fix for skim.Scanner so it will return a last line that does not
have a new line character.
- Drop zsio/json package- we're not doing this for now.

### CODE REVIEW

File: pkg/zsio/ndjson/parser.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+type Parser struct {
+	builder *zval.Builder
+	scratch []byte
+}
+
+func NewParser() *Parser {
+	return &Parser{builder: zval.NewBuilder()}
+}
+
+// Parse returns a new zson.Raw slice as well as an inferred zeek.Type
```
[the remark this answers is not in the document]
Comment (henridf) [reply pkg/zsio/ndjson/parser.go]: or better yet `zval.Encoding` ?

[the remark this answers is not in the document]
Comment (mattnibs) [reply pkg/zsio/ndjson/parser.go]: sho

File: pkg/zsio/ndjson/reader_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	r := ndjson.NewReader(strings.NewReader(input), resolver.NewTable())
+	require.NoError(t, zson.Copy(zson.NopFlusher(w), r))
+	NDJSONEq(t, input, out.String())
+}
+
+type output struct {
+	bytes.Buffer
+}
+
+func (o *output) Close() error { return nil }
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply pkg/zsio/ndjson/reader_test.go]: Yeah this is a relic of the Writers previously requiring a WriteCloser. Hard to keep up.

**Comment by mattnibs** [comment]
> Also this one! {"a": [{ "obj1": 1 },{ "obj2": 2 }]}

Yeah I talked with @mccanne and we're going to just reject lines that have multi-type vectors. I'll add functionality to do this. Something to revisit.

**Comment by mattnibs** [comment]
@henridf ready for another pass

**Pull request merged**